### PR TITLE
Bugfix: use spread notation to clone request object

### PR DESCRIPTION
**Pull request by bitpshr**
This pull request fixes an issue where the request object was being cloned incorrectly. Specifically, if prototype methods were added to a built-in object, like Array, these methods were being incorrectly copied to the newly-cloned object instance instead of its prototype. This caused upstream errors in MetaMask when `postMessage` was used to send array values since these arrays would have function members, which blows up structured cloning.

### CODE REVIEW

**Review by danfinlay — APPROVED** [review]
Wow, the spread operator as a less-buggy clone method. Great.

File: src/index.js [hunk 1]
```diff
@@ -30,7 +29,7 @@ class RpcEngine {
 
   _handle (_req, cb) {
     // clone request object
-    const req = clone(_req)
```
Comment (danjm) [inline src/index.js:33]: `clone` is a deep clone, spread is not. Is this a concern?

Comment (bitpshr) [reply src/index.js:33]: I think we should be OK: maintaining nested object references _after_ sending a transaction seems like something we don't need to support. Nested object values should just be able to be copied since these are just RPC method arguments.

**Review by danjm — APPROVED** [review]

**Comment by bitpshr** [comment]
Closing in favor of MetaMask/json-rpc-engine#1.

**Comment by bitpshr** [comment]
Reopening: we need to land this and cut a patch release here so this fix works properly.

Context: This is because several modules require `json-rpc-engine`, not just the extension. While we can pin the extension to the MetaMask-owned fork temporarily, we can't easily pin all dependencies that also use `json-rpc-engine` to that fork as well.

TLDR; we need to land this and cut a patch release. Then we can properly update the version as part of MetaMask/metamask-extension#5348. cc @kumavis

File: src/index.js (hunk 1 above)
Comment (kumavis) [reply src/index.js:33]: sounds like bugs and zalgo to me

Comment (kumavis) [reply src/index.js:33]: I wonder how JSON stringify reacts to their broken arrays

**Comment by kumavis** [comment]
switched to `Object.assign` so we dont introduce ecma2018
